crates/zed/src/zed/edit_prediction_registry.rs

```diff
@@ -3,14 +3,49 @@ use collections::HashMap;
 use copilot::{Copilot, CopilotCompletionProvider};
 use editor::Editor;
 use gpui::{AnyWindowHandle, App, AppContext as _, Context, Entity, WeakEntity};
+
 use language::language_settings::{EditPredictionProvider, all_language_settings};
-use settings::SettingsStore;
+use language_model::LanguageModelProvider;
+use language_models::{AllLanguageModelSettings, provider::ollama::OllamaLanguageModelProvider};
+use ollama;
+use ollama_edit_predictions::OllamaEditPredictionProvider;
+use settings::{Settings as _, SettingsStore};
 use std::{cell::RefCell, rc::Rc, sync::Arc};
 use supermaven::{Supermaven, SupermavenCompletionProvider};
 use ui::Window;
 use zeta::ZetaEditPredictionProvider;
 
 pub fn init(client: Arc<Client>, user_store: Entity<UserStore>, cx: &mut App) {
+    let ollama_provider = cx.new(|cx| OllamaLanguageModelProvider::new(client.http_client(), cx));
+
+    OllamaLanguageModelProvider::set_global(ollama_provider, cx);
+
+    // Authenticate the provider to ensure API key is loaded from keychain/environment.
+    // This is critical for the edit prediction provider to work correctly - without this,
+    // the OllamaEditPredictionProvider will not have access to the API key when making
+    // requests to the Ollama API, even though it correctly tries to retrieve the key
+    // from the global provider during request time. The authentication loads the API key
+    // from the system keychain or OLLAMA_API_KEY environment variable.
+    if let Some(provider) = OllamaLanguageModelProvider::global(cx) {
+        let api_url = language_models::provider::ollama::OllamaLanguageModelProvider::api_url(cx);
+        log::info!(
+            "Initializing Ollama edit predictions with API URL: {}",
+            api_url
+        );
+
+        if api_url.contains("ollama.com") {
+            log::info!(
+                "Detected Ollama Turbo configuration. Supported models: gpt-oss:20b, gpt-oss:120b, deepseek-v3.1:671b"
+            );
+            log::info!("Make sure you have a valid OLLAMA_API_KEY for Ollama Turbo access");
+        } else {
+            log::info!("Using local Ollama instance at: {}", api_url);
+        }
+
+        let task = provider.update(cx, |provider, cx| provider.authenticate(cx));
```

Comment:
Ollama authenticate should not be called on zed init, only if it's actually be configured to be used.